src/java.base/share/classes/java/util/concurrent/ForkJoinWorkerThread.java

```diff
@@ -84,7 +80,7 @@ public class ForkJoinWorkerThread extends Thread {
         super.setDaemon(true);
         if (handler != null)
             super.setUncaughtExceptionHandler(handler);
-        if (useSystemClassLoader)
+        if (useSystemClassLoader & !clearThreadLocals) // else done by Thread ctor
```

Comment:
It might be clearer to use a conditional-and here although it doesn't make any difference here.

Reply:
I agree that "&&" would be clearer. Will do.